optaplanner-examples/src/test/java/org/optaplanner/examples/conferencescheduling/solver/ConferenceSchedulingScoreRulesXlsxTest.java

```diff
@@ -0,0 +1,228 @@
+            scoreVerifier.assertSoftWeight(currentPackage, currentConstraint, expectedScore.getSoftScore(), currentSheetSolution);
+        }
+    }
+
+    private class testConferenceSchedulingScoreRulesReader extends AbstractXlsxSolutionFileIO.AbstractXlsxReader<ConferenceSolution> {
+
+        private final SolutionCloner<ConferenceSolution> solutionCloner =
+                SolutionDescriptor.buildSolutionDescriptor(ConferenceSolution.class, Talk.class).getSolutionCloner();
+
+        private int numberOfSheets, currentTestSheetIndex;
+        private Map<String, Room> roomMap;
+        private Map<Pair<LocalDateTime, LocalDateTime>, Timeslot> timeslotMap;
+        private Map<Integer, LocalDate> columnIndexToDateMap;
+        private Map<Integer, LocalTime> columnIndexToStartTimeMap;
+        private Map<Integer, LocalTime> columnIndexToEndTimeMap;
+
+        private testConferenceSchedulingScoreRulesReader(XSSFWorkbook workbook) {
+            super(workbook);
+            this.numberOfSheets = workbook.getNumberOfSheets();
+            this.currentTestSheetIndex = workbook.getSheetIndex("Talks") + 1;
+            roomMap = initialSolution.getRoomList().stream().collect(
+                    Collectors.toMap(Room::getName, Function.identity()));
+            timeslotMap = initialSolution.getTimeslotList().stream().collect(
+                    Collectors.toMap(timeslot -> Pair.of(timeslot.getStartDateTime(), timeslot.getEndDateTime()),
+                            Function.identity()));
+            this.columnIndexToDateMap = new HashMap<>(timeslotMap.size());
+            this.columnIndexToStartTimeMap = new HashMap<>(timeslotMap.size());
+            this.columnIndexToEndTimeMap = new HashMap<>(timeslotMap.size());
+        }
+
+        @Override
+        public ConferenceSolution read() {
+            return initialSolution;
+        }
+
+        private ConferenceSolution nextSolution() {
+
+            if (currentTestSheetIndex >= numberOfSheets) {
+                return null;
+            }
+
+            nextSheet(workbook.getSheetName(currentTestSheetIndex++));
+
+            nextRow(false);
+            readHeaderCell("Constraint package");
+            currentPackage = nextStringCell().getStringCellValue();
+            nextRow(false);
+            readHeaderCell("Constraint name");
+            currentConstraint = nextStringCell().getStringCellValue();
+            nextRow(false);
+            readHeaderCell("Score");
+            expectedScore = HardSoftScore.parseScore(nextStringCell().getStringCellValue());
+
+            ConferenceSolution nextSheetSolution = solutionCloner.cloneSolution(initialSolution);
+            Map<String, Talk> talkMap = nextSheetSolution.getTalkList().stream().collect(
+                    Collectors.toMap(Talk::getCode, Function.identity()));
+
+            logger.info(currentSheet.getSheetName());
```

Comment:
if we want to keep this log message (it's not too verbose for running the tests in "mvn clean install") then it's better to have something like 
  logger.info("  Reading sheet ({}).", currentSheet.getSheetName());